stdlib.sh

```diff
@@ -804,6 +804,13 @@ layout_node() {
   PATH_add node_modules/.bin
 }
 
+# Usage: layout opam
+#
+# Sets environment variables from `opam env`.
+layout_opam() {
+  eval "$(opam env)"
```

Comment:
What files does opam load to generate that env?

Adding a `watch_file` before the eval is useful, so direnv automatically reloads whenever the opam env changes.

Reply:
@zimbatm What `opam env` returns is not actually a file, but a set of environment variables!

Reply:
Is there a per-project config file that Opam uses to figure out what version of Ocaml to use, for example?

Reply:
It could be `*.opam` file(s), but there is no equivalent of the `.node-version` file in Node.js ecosystem in the OCaml ecosystem.